server/src/file-storage/storage-providers/local-storage.ts

```diff
@@ -0,0 +1,30 @@
+import { FileStorageArgs, IFileStorage } from '../file-storage.service';
+import { createWriteStream } from 'fs';
+import * as path from 'path';
+
+export class LocalStorageProvider implements IFileStorage {
+  public uploadFile = async (uploadArgs: FileStorageArgs): Promise<string> => {
+    const { file, prefix, userId } = uploadArgs;
+    const readableStream = uploadArgs.file.buffer;
+    const fileName = `${prefix}-${userId}.${file.originalname.split('.').pop()}`;
+    const devFileUrl = `http://localhost:3000/${fileName}`;
+    const writableStream = createWriteStream(
+      path.join(__dirname, '..', '..', '..', '..', 'client', 'public', fileName),
+      { flags: 'a' }, // append files
+    );
+
+    writableStream.on('end', () => {
+      console.log(`############failed to save: ${fileName}`);
+    });
+
+    writableStream.on('error', (err) => {
+      console.log(err.stack);
+    });
+
+    writableStream.write(readableStream);
+
+    return Promise.resolve(devFileUrl);
+  };
+}
```

Comment:
This is used in development and it saves files as React assets, which are accessible by `http://localhost:3000/userprofile-{user_id}.{image_extension}`